### PR TITLE
Fix bug when enabling dask-expr

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

Yikes, I didn't handle the None case properly.

### CODE REVIEW

File: dask/dataframe/__init__.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ def _dask_expr_enabled() -> bool:
             import dask_expr  # noqa: F401
         except ImportError:
             raise ValueError("Must install dask-expr to activate query planning.")
+        use_dask_expr = True
     return use_dask_expr
```
Comment (hendrikmakait) [inline dask/dataframe/__init__.py:21]: I _think_ this is what we want, i.e., try to use `dask-expr` if enabled **and** `pandas>=2` is installed:

```suggestion
        return True
    return False
```

Comment (hendrikmakait) [reply dask/dataframe/__init__.py:21]: Alternatively, you could also rewrite this to return early with 
```python
    use_dask_expr = dask.config.get("dataframe.query-planning")
    if use_dask_expr is False or use_dask_expr is None and Version(pd.__version__).major < 2:
        return False
    try:
        import dask_expr  # noqa: F401
    except ImportError:
        raise ValueError("Must install dask-expr to activate query planning.")
    return True
```

Comment (phofl) [reply dask/dataframe/__init__.py]: I like the second version, added

File: docs/source/conf.py [hunk 2]
```diff
@@ -14,6 +14,10 @@
 
 import os
 
+# Build docs with option disabled until we move dask-expr over, othwise
```
Comment (hendrikmakait) [inline docs/source/conf.py:17]:
```suggestion
# Build docs with option disabled until we move dask-expr over, otherwise
```

**Review by hendrikmakait — APPROVED** [review]
Thanks, @phofl! `test_to_parquet_overwrite_files_from_read_parquet_in_same_call_raises` isn't happy on Windows, but I consider that non-blocking.

**Pull request merged**